heron/proto/ckptmgr.proto

```diff
@@ -171,7 +175,26 @@ message CleanStatefulCheckpointResponse {
   repeated string cleaned_checkpoint_ids = 2;
 }
 
-// stmgr -> ckptmgr messages
+/*
+ * stmgr -> ckptmgr messages
+ */
+
+// This message encapsulates the info associated with
+// state of an instance/partition
+message InstanceStateCheckpoint {
+  required string checkpoint_id = 1;
+  required bytes state = 2;
+  // A version string that can be specified by user. It can be used to translate
```

Comment:
exactly what does this data version do?

Reply:
We are thinking about calculation change. Not sure how useful it is though. Please feel free to let us know if you feel this is overkill.

Normally the calculation in bolts won't change much and the version number doesn't need to change. However in case the equation is changed, for example,  the old value is a duration in seconds and it is changed to duration in millisecond, or it is possible the value is a data structure and a new field is added. User can increase the version number and provide a conversion function between the two versions. Then the topology would be able to load the stateful data stored before the update, or skip/reject the data if topology doesn't know how to convert between the two versions.

If topology cant tell the difference and just load the data, the state could be polluted without user knowing it. The current design seems to rely on users manually removing the data? it could be risky and hard since the topology needs to be running in order to do it I think?

Reply:
IMO this really pollutes the abstraction. You have now introduced some kind of versioning in the system at such a low level to begin with without any kind of higher level api/rationale.

Reply:
kk. Let me remove it for now until we have a better idea/support.